tests/FSharp.Compiler.ComponentTests/Language/DiscriminatedUnionTests.fs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.ComponentTests.Language
+
+open Xunit
+open FSharp.Test.Utilities.Compiler
+open FSharp.Quotations.Patterns
+
+module DiscriminatedUnionTests =
+
+    [<Fact>]
+    let ``Is* discriminated union properties are visible`` () =
+        FSharp """
+namespace rec Hello
+
+[<Struct>]
+type Foo =
+    private
+    | Foo of string
+    | Bar
+
+module Main =
+    [<EntryPoint>]
+    let main _ =
+        let foo = Foo.Foo "hi"
+        printfn "IsFoo: %b / IsBar: %b" foo.IsFoo foo.IsBar
```

Comment:
It would be good to have tests

1. that these are present in autocomplete lists
2. what the results of "go to definition" on these is
3. what happens when you try to rename this symbol (I guess it shouldn't be allowed)

I suspect (1) and (2) will be ok but (3) will not be - though I think we don't mind that much what happens with (3)

Ideally (1) and (2) should have automated tests